tfmesos/scheduler.py

```diff
@@ -359,23 +372,55 @@ def registered(self, driver, framework_id, master_info):
             )
 
     def statusUpdate(self, driver, update):
-        mesos_task_id = int(update.task_id.value)
-        if update.state != 'TASK_RUNNING':
+        logger.debug('Received status update %s', str(update.state))
+        mesos_task_id = update.task_id.value
+        if self._is_terminal_state(update.state):
             task = self.tasks[mesos_task_id]
```

Comment:
That can happen only if mesos sends multiple terminal state (finished, failed etc ) right ?